web/packages/teleport/src/Users/UserDetails/UserDetails.tsx

```diff
@@ -0,0 +1,313 @@
+                    <Text color="text.muted">View Locks</Text>
+                    <Icons.ArrowRight color="text.muted" size="small" />
+                  </Button>
+                )}
+              </Flex>
+            </UserDetailField>
+          </UserDetailsGrid>
+        </InfoParagraph>
+      </UserDetailsSection>
+
+      {sections.map((SectionComponent, index) => (
+        <UserDetailsSection key={index}>
+          <SectionComponent user={user} onEdit={onEdit} />
+        </UserDetailsSection>
+      ))}
+    </Box>
+  );
+}
+
+const UserDetailsActions = ({
+  user,
+  onEdit,
+  onReset,
+  onDelete,
+}: {
+  user: User;
+  onEdit?: () => void;
+  onReset?: () => void;
+  onDelete?: () => void;
+}) => {
+  if (!(onEdit || onDelete)) {
+    return null;
+  }
+
+  if (user.isBot || !user.isLocal) {
+    return null;
+  }
+
+  return (
+    <MenuIcon
+      Icon={MoreVert}
+      menuProps={{
+        anchorOrigin: {
+          vertical: 'bottom',
+          horizontal: 'right',
+        },
+        transformOrigin: {
+          vertical: 'top',
+          horizontal: 'right',
+        },
+      }}
+    >
+      {onEdit && <MenuItem onClick={onEdit}>Edit</MenuItem>}
+      {onReset && <MenuItem onClick={onReset}>Reset Authentication</MenuItem>}
+      {onDelete && <MenuItem onClick={onDelete}>Delete</MenuItem>}
+    </MenuIcon>
+  );
+};
+
+export interface UserDetailsTitleProps {
+  user: User;
+  onEdit?: () => void;
+  onReset?: () => void;
+  onDelete?: () => void;
+  panelWidth?: number;
+}
+
+export function UserDetailsTitle({
+  user,
+  onEdit,
+  onReset,
+  onDelete,
+  panelWidth = 480,
+}: UserDetailsTitleProps) {
+  const { text: authType, icon } = renderAuthType(user);
+
+  // needed to fill InfoGuidePanel for UserDetailsActions
+  const containerWidth = panelWidth - 80;
+  const userIconSize = 48;
+
+  return (
+    <Flex
+      alignItems="center"
+      justifyContent="space-between"
+      minWidth={`${containerWidth}px`}
+    >
+      <Flex alignItems="center" gap={3}>
+        {user.isBot ? (
+          <Icons.Bots size={userIconSize} />
+        ) : (
+          <Icons.User size={userIconSize} />
+        )}
+        <Box maxWidth={containerWidth - 96}>
+          <Text fontSize={3} fontWeight="bold" title={user.name}>
+            {user.name}
+          </Text>
+          <Flex alignItems="center" gap={2}>
+            <ResourceIcon name={icon} width="16px" height="16px" />
+            <Text
+              fontSize={2}
+              color="text.muted"
+              fontWeight="normal"
+              style={{ textTransform: 'capitalize' }}
+            >
+              {authType}
+              {user.isBot ? ' (Bot)' : ''}
+            </Text>
+          </Flex>
+        </Box>
+      </Flex>
+      <UserDetailsActions
+        user={user}
+        onEdit={onEdit}
+        onReset={onReset}
+        onDelete={onDelete}
+      />
+    </Flex>
+  );
+}
+
+function renderAuthType(user: User): AuthTypeInfo {
```

Comment:
It was a bit confusing to find a "render" function that didn't actually render a React node, so consider renaming it. How about `toAuthTypeInfo`?